### PR TITLE
net-p2p/syncthing: fix relaysrv service files

### PR DESCRIPTION
**Pull request by khumarahn**
@sbraz 

I will be travelling for a couple of weeks. May not be able to respond.

### CODE REVIEW

**Comment by sbraz** [comment]
You should rebase these commits into one. Also, why don't we at least keep the same username/groupe ?

**Comment by khumarahn** [comment]
If we keep user:group, then we need to patch upstream's systemd service file. And keep track of it forever. And it will be more difficult to make changes later, when more users come. It is easier to stay with upstream while only very few of us could be affected.

**Comment by sbraz** [comment]
OK, I didn't realise the unit had the usernames hardcoded. Maybe we need to publish a news item or something though…

**Comment by khumarahn** [comment]
Squashed. I am not sure it is worth a news item.

**Comment by monsieurp** [comment]
Publish a warning / message via elog if you need to let the user know of a special case with the update.

**Comment by sbraz** [comment]
We could, yes: if the old user exists, warn that it was renamed.

File: net-p2p/syncthing/syncthing-0.14.0-r1.ebuild [hunk 1]
```diff
@@ -86,14 +86,11 @@ src_install() {
 	if use tools ; then
 		# openrc and systemd service files
 		systemd_dounit "${S}"/src/${EGO_PN}/cmd/strelaysrv/etc/linux-systemd/strelaysrv.service
-		newconfd "${FILESDIR}/${PN}-relaysrv.confd" ${PN}-relaysrv
-		newinitd "${FILESDIR}/${PN}-relaysrv.initd" ${PN}-relaysrv
-
-		keepdir /var/lib/${PN}-relaysrv
```
Comment (sbraz) [inline net-p2p/syncthing/syncthing-0.14.0-r1.ebuild]: As @djc was asking on IRC: why aren't these needed any more? Was it an omission?

Comment (khumarahn) [inline net-p2p/syncthing/syncthing-0.14.0-r1.ebuild]: Regards to @djc. These lines seem to have been wrong, and not needed. `/var/lib/strelaysrv` is a home dir for `strelaysrv` by design. We don't install any files to keep there. Correct ownership on `/var/log/syncthing/strelaysrv.log` is ensured by the openrc init script, while systemd does not use it.

Comment (khumarahn) [inline net-p2p/syncthing/syncthing-0.14.0-r1.ebuild]: I figured out I was wrong removing those lines. That is, they were incorrect, but  `enewuser` does not create the home directory for the new user by default.

**Comment by khumarahn** [comment]
Added old user warning. Would be nice if someone could check.

**Comment by sbraz** [comment]
You should probably use the `user`eclass if it's supported bi EAPI 6. I see packages using it like so:
`if [ -z "$(egetent passwd bareos 2>/dev/null)" ]; then …` (`if … then`  seems easier than going with a bash subshell too).
As for the message, I'd make it a bit simpler (and use plural everywhere):
```sh
		ewarn "The user and group for the relay server have been changed"
		ewarn "from syncthing-relaysrv to strelaysrv"
		ewarn "The old user and group are not deleted automatically. Delete them by running:"
		ewarn "    userdel -r syncthing-relaysrv"
		ewarn "    groupdel syncthing-relaysrv"
```
Also, I think the indentation is off with the first `ewarn`.

**Comment by khumarahn** [comment]
ok, I agree. I replaced the last commit with your corrections.

**Comment by djc** [comment]
Rebased and pushed as 5210bac33931b2b87, thanks!

**Pull request closed without merging**